service-config/src/main/java/com/palantir/config/service/ServiceConfiguration.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.config.service;
+
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.google.common.base.Optional;
+import com.palantir.remoting.ssl.SslConfiguration;
+import com.palantir.tokens.auth.BearerToken;
+import java.util.List;
+import org.immutables.value.Value.Immutable;
+import org.immutables.value.Value.Style;
+
+@Immutable
+@JsonDeserialize(as = ImmutableServiceConfiguration.class)
+@Style(visibility = Style.ImplementationVisibility.PACKAGE)
+public abstract class ServiceConfiguration {
+
+    public abstract Optional<BearerToken> apiToken();
+
+    public abstract Optional<SslConfiguration> security();
+
+    public abstract List<String> uris();
```

Comment:
Using Set due to FeignClientFactory and that is impl detail. We want use list so we can specify priorities in which the urls will be used.